textplot/utils.py

```diff
@@ -3,13 +3,83 @@
 import re
 import numpy as np
 import functools
+import copy
 
 from collections import OrderedDict
 from nltk.stem import PorterStemmer
 from itertools import islice
 
+import spacy
+from nltk.stem.snowball import SnowballStemmer
 
-def tokenize(text):
+class SpacyTokenizer(object):
+    def __init__(self, lang='de', lower=False):
+
+        if lang == 'de':
+            spacy_model = 'de_core_news_sm'
+        elif lang == 'en':
+            spacy_model = 'en_core_web_sm'
+        else:
+            print(f'Selected language not yet supported: {lang}')
+            raise NotImplementedError
+        print(f'Initializing tokenizer: Spacy tokenizer with model {spacy_model}')
+        self.nlp = spacy.load(spacy_model)
+        self.lower = lower
+
+        self.global_offset = 0
+
+    def tokenize(self, collection):
+
```

Comment:
It's recommended to use spacy's `nlp.pipe(texts)` for running on large collections of texts. However, I think it's fine to leave this as is for the time being and focus on optimisation later.